driver-mapping/src/main/java/com/datastax/driver/mapping/MapperConfiguration.java

```diff
@@ -0,0 +1,336 @@
+         *
+         * @return whether or not the strategy allows scanning of parent classes
+         */
+        public boolean isHierarchyScanEnabled() {
+            return hierarchyScanEnabled;
+        }
+
+        /**
+         * Sets whether or not the strategy allows scanning of parent classes
+         *
+         * @param hierarchyScanEnabled whether or not the strategy allows scanning of parent classes
+         * @return the HierarchyScanStrategy to enable builder pattern
+         */
+        public HierarchyScanStrategy setHierarchyScanEnabled(boolean hierarchyScanEnabled) {
+            this.hierarchyScanEnabled = hierarchyScanEnabled;
+            return this;
+        }
+
+        /**
+         * Returns whether or not the strategy allows scanning of all parent classes or only those
+         * annotated with {@code Table} or {@code UDT} or {@code Accessor}
+         *
+         * @return whether or not the strategy allows scanning of all parent classes or only those
+         * annotated with {@code Table} or {@code UDT} or {@code Accessor}
+         */
+        public boolean isScanOnlyAnnotatedClasses() {
+            return scanOnlyAnnotatedClasses;
+        }
+
+        /**
+         * Sets whether or not the strategy allows scanning of all parent classes or only those
+         * annotated with {@code Table} or {@code UDT} or {@code Accessor}
+         *
+         * @param scanOnlyAnnotatedClasses whether or not the strategy allows scanning of
+         *                                 all parent classes or only those annotated with
+         *                                 {@code Table} or {@code UDT} or {@code Accessor}
+         * @return the HierarchyScanStrategy to enable builder pattern
+         */
+        public HierarchyScanStrategy setScanOnlyAnnotatedClasses(boolean scanOnlyAnnotatedClasses) {
+            this.scanOnlyAnnotatedClasses = scanOnlyAnnotatedClasses;
+            return this;
+        }
+
+        /**
+         * Returns the deepest parent class to the strategy allows to scan
+         * (which default to {@code Object.class})
+         *
+         * @return the deepest parent class to the strategy allows to scan
+         * (which default to {@code Object.class})
+         */
+        public Class<?> getDeepestAllowedAncestor() {
+            return deepestAllowedAncestor;
+        }
+
+        /**
+         * Sets the deepest parent class to the strategy allows to scan
+         *
+         * @param deepestAllowedAncestor the deepest parent class to the strategy allows to scan
+         * @return the HierarchyScanStrategy to enable builder pattern
+         */
+        public HierarchyScanStrategy setDeepestAllowedAncestor(Class<?> deepestAllowedAncestor) {
+            this.deepestAllowedAncestor = deepestAllowedAncestor;
+            return this;
+        }
+
+    }
+
+    public enum PropertyMappingStrategy {BLACK_LIST, WHITE_LIST}
```

Comment:
I would prefer `OPT_IN` and `OPT_OUT`, but if I'm the only one to ask that, it's fine to keep it the way it is.